src/libs/utils.js

```diff
@@ -369,6 +369,29 @@ export const truncateInteger = integer => {
   return `${Math.floor(integer / 1000)}k`
 }
 
+/**
+ * Polls using a given function until the pollUntil function returns true.
+ * @param {() => { result: A, shouldContinue: boolean }} pollFn - The function to poll using
+ * @param {number} pollTime - How much time there should be in ms between calls of the pollFn
+ * @param {boolean} leading - Whether the function should wait {pollTime} ms before running for the first time
+
+ * @returns {A} - The response of pollFn
+ */
+export const poll = async (pollFn, pollTime, leading = true) => {
+  do {
```

Comment:
I'm surprised...do we not have a polling function already?  Like, for the import service, does it not use one already?

Reply:
We have a `usePollingEffect`, but that is component driven, not webpage wide. (Unless the component is top level)

I created this as a need to have a webpage wide poll while we were doing the async approach, and think it provides enough value to leave in the PR.

Reply:
ah cool.  Yeah, that makes sense